### ISSUE TITLE
goconst: hardcoded IgnoreTests: false is not equivalent to exclusion rules

### ISSUE THREAD

**Issue by franciscocpg** [golangci/golangci-lint#6479]
### Welcome

- [x] Yes, I'm using a binary release within 2 latest [releases](https://github.com/golangci/golangci-lint/releases). Only such installations are supported.
- [x] Yes, I've searched similar [issues on GitHub](https://github.com/golangci/golangci-lint/issues) and didn't find any.
- [x] Yes, I've read the `typecheck` section of the [FAQ](https://golangci-lint.run/docs/welcome/faq/#why-do-you-have-typecheck-errors).
- [x] Yes, I've tried with the standalone [linter](https://golangci-lint.run/docs/linters/) if available (e.g., gocritic, go vet, etc.).
- [x] I agree to follow this project's [Code of Conduct](https://github.com/golangci/golangci-lint?tab=coc-ov-file)

### How did you install golangci-lint?

go install

### Description of the problem

In [`pkg/golinters/goconst/goconst.go`](https://github.com/golangci/golangci-lint/blob/b3ac139dffb84c6e541422b0f73252e7a29313fc/pkg/golinters/goconst/goconst.go#L63-L64), `IgnoreTests` is hardcoded to `false` with the comment:

```go
// Should be managed with `linters.exclusions.rules`.
IgnoreTests: false,
```

The assumption is that users can use `linters.exclusions.rules` with `path: _test\.go` to achieve the same effect. **This is incorrect** — the two mechanisms are not equivalent, leading to non-deterministic (flaky) CI failures.

## Why they're not equivalent

1. With `IgnoreTests: false`, goconst counts occurrences across **all** files in the package (test + non-test)
2. goconst picks **one** file location to report the issue — the location depends on the order of `pass.Files` from `go/analysis`, which is non-deterministic across platforms (e.g., darwin/arm64 vs linux/amd64)
3. The exclusion rule `path: _test\.go` only checks whether that **single report location** matches

So if a string like `"email"` appears 2 times in `job.go` and 3 times in `job_test.go`:
- goconst finds 5 total occurrences (above the default threshold of 3) and reports **one** issue
- On **macOS**: goconst happens to report at `job_test.go` → exclusion rule matches → ✅ no error
- On **Linux CI**: goconst happens to report at `job.go` → exclusion rule doesn't match → ❌ flaky failure

With `IgnoreTests: true`, goconst would only count the 2 occurrences in `job.go`, which is below the threshold — no issue is generated at all.

### Version of golangci-lint

<details>

```console
$ golangci-lint --version
golangci-lint has version v2.8.0-custom-gcl-TarcVizjLG9v7VgBsBKqKYyHFGi157eTgVICTwxfmqY built with go1.25.1 from ? on 2026-04-02 13:43:14.117729 +0000 UTC
```

</details>

### Configuration

<details>

```yaml
version: "2"
linters:
  enable:
    - goconst
  exclusions:
    rules:
      - linters:
          - goconst
        path: _test\.go
```

</details>

### Go environment

<details>

```console
$ go version && go env
AR='ar'
CC='clang'
CGO_CFLAGS='-O2 -g'
CGO_CPPFLAGS=''
CGO_CXXFLAGS='-O2 -g'
CGO_ENABLED='1'
CGO_FFLAGS='-O2 -g'
CGO_LDFLAGS='-O2 -g'
CXX='clang++'
GCCGO='gccgo'
GO111MODULE=''
GOARCH='arm64'
GOARM64='v8.0'
GOAUTH='netrc'
GOBIN='/Users/francisco.guimaraes/.local/share/mise/installs/go/1.25.1/bin'
GOCACHE='/Users/francisco.guimaraes/Library/Caches/go-build'
GOCACHEPROG=''
GODEBUG=''
GOENV='/Users/francisco.guimaraes/Library/Application Support/go/env'
GOEXE=''
GOEXPERIMENT=''
GOFIPS140='off'
GOFLAGS=''
GOGCCFLAGS='-fPIC -arch arm64 -pthread -fno-caret-diagnostics -Qunused-arguments -fmessage-length=0 -ffile-prefix-map=/var/folders/yn/_cvr7m5x3pdcq9klvz48dw640000gp/T/go-build3235736671=/tmp/go-build -gno-record-gcc-switches -fno-common'
GOHOSTARCH='arm64'
GOHOSTOS='darwin'
GOINSECURE=''
GOMOD='/dev/null'
GOMODCACHE='/Users/francisco.guimaraes/go/pkg/mod'
GONOPROXY='github.com/jusbrasil'
GONOSUMDB='github.com/jusbrasil'
GOOS='darwin'
GOPATH='/Users/francisco.guimaraes/go'
GOPRIVATE='github.com/jusbrasil'
GOPROXY='https://proxy.golang.org,direct'
GOROOT='/Users/francisco.guimaraes/.local/share/mise/installs/go/1.25.1'
GOSUMDB='sum.golang.org'
GOTELEMETRY='local'
GOTELEMETRYDIR='/Users/francisco.guimaraes/Library/Application Support/go/telemetry'
GOTMPDIR=''
GOTOOLCHAIN='auto'
GOTOOLDIR='/Users/francisco.guimaraes/.local/share/mise/installs/go/1.25.1/pkg/tool/darwin_arm64'
GOVCS=''
GOVERSION='go1.25.1'
GOWORK='/Users/francisco.guimaraes/code/github.com/jusbrasil/go-monorepo/go.work'
PKG_CONFIG='pkg-config'
```

</details>

### Verbose output of running

<details>

```console
$ golangci-lint cache clean
$ golangci-lint run -v
# paste output here
```

</details>

### A minimal reproducible example or link to a public repository

<details>

```go
// add your code here
```

</details>

### Validation

- [x] Yes, I've included all information above (version, config, etc.).

### Supporter

- [ ] I am a [sponsor/backer](https://donate.golangci.org) of this project.

**Comment by franciscocpg** [comment]
Hi @ldez 

Why was it closed?
